### ISSUE TITLE
MetaProperty should have two Identifiers

### ISSUE THREAD

**Issue by mysticatea** [eslint/espree#206]
ESTree says MetaProperty has two Identifiers: https://github.com/estree/estree/blob/master/es6.md#metaproperty
But `espree` is parsing to two string properties.

**Code:**

```js
function foo() { new.target; }
```

**Actual:**

```js
{
  "type": "Program",
  "body": [
    {
      "type": "FunctionDeclaration",
      "id": {
        "type": "Identifier",
        "name": "foo",
        "range": [
          9,
          12
        ]
      },
      "params": [],
      "body": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "MetaProperty",
              "meta": "new",
              "property": "target",
              "range": [
                17,
                27
              ]
            },
            "range": [
              17,
              28
            ]
          }
        ],
        "range": [
          15,
          30
        ]
      },
      "generator": false,
      "expression": false,
      "range": [
        0,
        30
      ]
    }
  ],
  "sourceType": "script",
  "range": [
    0,
    30
  ]
}
```

**Expected:**

```js
{
  "type": "Program",
  "body": [
    {
      "type": "FunctionDeclaration",
      "id": {
        "type": "Identifier",
        "name": "foo",
        "range": [
          9,
          12
        ]
      },
      "params": [],
      "body": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "MetaProperty",
              "meta": {
                "type": "Identifier",
                "name": "new",
                "range": [
                  17,
                  19
                ]
              },
              "property": {
                "type": "Identifier",
                "name": "target",
                "range": [
                  21,
                  27
                ]
              },
              "range": [
                17,
                27
              ]
            },
            "range": [
              17,
              28
            ]
          }
        ],
        "range": [
          15,
          30
        ]
      },
      "generator": false,
      "expression": false,
      "range": [
        0,
        30
      ]
    }
  ],
  "sourceType": "script",
  "range": [
    0,
    30
  ]
}
```

**Comment by nzakas** [comment]
Yup, this will be fixed when we switch to Acorn

**Comment by nzakas** [comment]
Working on this

**Comment by nzakas** [comment]
Verified this is working on master and a test already exists.
